spec/account_spec.rb

```diff
@@ -31,180 +31,209 @@
-      subject.messages.should have(1).items
+      expect(subject.messages.count).to be 1
     end
   end
 
   describe "#send_messages" do
     let(:batch_id) { random_string }
-    let(:uri_prefix) { "https://api.esendex.com/messages/" }
+    let(:uri_prefix) { 'https://api.esendex.com/messages/' }
     let(:message_one_id) { random_guid }
     let(:message_two_id) { random_guid }
     let(:send_response_xml) {
-      "<?xml version=\"1.0\" encoding=\"utf-8\"?> 
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>
       <messageheaders batchid=\"#{batch_id}\" xmlns=\"http://api.esendex.com/ns/\">
         <messageheader\ uri=\"#{uri_prefix}#{message_one_id}\" id=\"#{message_one_id}\" />
         <messageheader\ uri=\"#{uri_prefix}#{message_two_id}\" id=\"#{message_two_id}\" />
       </messageheaders>"
     }
 
     before(:each) do
-      api_connection.stub(:post) { mock('Response', :body => send_response_xml) }
+      allow(api_connection).to receive(:post).and_return( double('Response', body: send_response_xml) )
     end
 
     subject { account.send_message(to: "447815777555", body: "Hello from the Esendex Ruby Gem") }
 
     it "should post to the message dispatcher resource" do
-      api_connection.should_receive(:post).with("/v1.0/messagedispatcher", anything)
+      expect(api_connection).to receive(:post).with("/v1.0/messagedispatcher", anything)
       subject
     end
-    
+
     it "should provide a list containing two messages" do
-      subject.messages.should have(2).items
+      expect(subject.messages.count).to be 2
     end
-    
+
     it "should have message one in the message list" do
-      subject.messages.should include(id: "#{message_one_id}", uri: "#{uri_prefix}#{message_one_id}")
+      expect(subject.messages).to include(id: "#{message_one_id}", uri: "#{uri_prefix}#{message_one_id}")
     end
-    
+
     it "should have message two in the message list" do
-      subject.messages.should include(id: "#{message_two_id}", uri: "#{uri_prefix}#{message_two_id}")
+      expect(subject.messages).to include(id: "#{message_two_id}", uri: "#{uri_prefix}#{message_two_id}")
+    end
+  end
+
+  describe "#message_status" do
+    let(:uri_prefix) { "https://api.esendex.com/v1.0/messageheaders/" }
+    let(:message_id) { random_guid }
+    let(:phonenumber) { "12345678" }
+    let(:status) { "Acknowledged" }
+    let(:send_response_xml) {
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>
+      <messageheader id=\"#{message_id}\" xmlns=\"http://api.esendex.com/ns/\" uri=\"#{uri_prefix}#{message_id}\">
+        <status>#{status}</status>
+        <to>
+          <phonenumber>#{phonenumber}</phonenumber>
+        </to>
+      </messageheader>"
+    }
+
+    before(:each) do
+      allow(api_connection).to receive(:get).and_return( double('Response', body: send_response_xml) )
+    end
+
+    subject { account.message_status(message_id) }
+
+    it "should post to the message dispatcher resource" do
```

Comment:
Can you rename this test to match the case. Something like "should get the messageheader resource with a message id"